controllers/build-tool-detector.go

```diff
@@ -42,22 +49,49 @@ const (
 type BuildToolDetectorController struct {
 	*goa.Controller
 	config.Configuration
+	httpClient *http.Client
 }
 
 // NewBuildToolDetectorController creates a build-tool-detector controller.
 func NewBuildToolDetectorController(service *goa.Service, configuration config.Configuration) *BuildToolDetectorController {
-	return &BuildToolDetectorController{Controller: service.NewController(buildToolDetectorController), Configuration: configuration}
+	return &BuildToolDetectorController{
+		Controller: service.NewController(buildToolDetectorController),
+		Configuration: configuration,
+		httpClient:  http.DefaultClient,
+	}
 }
 
 // Show runs the show action.
 func (c *BuildToolDetectorController) Show(ctx *app.ShowBuildToolDetectorContext) error {
 	rawURL := ctx.URL
-	repositoryService, err := repository.CreateService(rawURL, ctx.Branch, c.Configuration)
 	ctx.ResponseWriter.Header().Set(contentType, applicationJSON)
+
+	// Get auth service to retrieve GH's user token
+	u, err := url.Parse(c.Configuration.GetAuthServiceURL())
+	if err != nil {
+		return handleError(ctx, err)
+	}
+	authClient := client.New(goaclient.HTTPClientDoer(c.httpClient))
+	authClient.Host = u.Host
+	authClient.Scheme = u.Scheme
+	if goajwt.ContextJWT(ctx) != nil {
+		authClient.SetJWTSigner(goasupport.NewForwardSigner(ctx))
+	} else {
+		log.Logger().Info(ctx, nil, "no token in context")
+	}
+	tr := tokenRetriever{authClient: authClient, context: ctx}
+	scm, err := c.getScm(ctx)
```

Comment:
I don't think we should set a default scm. The reason's we wouldn't want to do this are the api becomes a lot less fluid. Our service is trapped to just supporting one type of scm while running. Also, if it is not set, we automatically assume it is github and if it ever happens to be something else, the application will fail.  

Ideally, the application should analyze the url provided and figure out what the scm is. There shouldn't be a default - it is either an scm we support and know about or not.

At this point in the code, we haven't analyzed the url and hence we don't quite know what type of scm we are dealing with.